certbot-ci/certbot_integration_tests/utils/acme_server.py

```diff
@@ -192,9 +192,6 @@ def main():
                   .format(acme_xdist['directory_url']))
             print('--> Press CTRL+C to stop the ACME server.')
 
-            docker_name = 'pebble_pebble_1' if 'pebble' in server_type else 'boulder_boulder_1'
-            process = subprocess.Popen(['docker', 'logs', '-f', docker_name])
```

Comment:
What do you think about adding code like this to the case when `ACMEServer` is used with boulder and stdout is printed? Currently in this PR, `run_acme_server` produces no additional output once boulder has been started.

Reply:
Are you sure you are on the latest version of my PR? I pass now the `docker-compose up` through `self._launch_process`, and so see the GIT, docker-compose and docker processes output in my terminal when running `run_acme_server boulder-v2` for instance.

Reply:
Oh, I got it. In fact you will not get more log from it by running `docker-compose logs`. All dockers logs are already displayed by `docker-compose up` since it is not run in detached mode.

Reply:
Is that right?

In one terminal, I ran `run_acme_server boulder-v2`. The last output it ever printed while it was running was:
```
boulder_1    | GOBIN=/go/src/github.com/letsencrypt/boulder/bin GO111MODULE=on go install -mod=vendor  ./...
=> Finished boulder instance deployment.
--> Instance of http://localhost:4001/directory is running, directory URL is http://localhost:4001/directory
--> Press CTRL+C to stop the ACME server.
```
In a second terminal I ran:
```
docker logs -f boulder_boulder_1
```
which continued to produce output as I ran Certbot against Boulder.

Reply:
This is very surprising. I always use docker-compose up without detached mode to get the logs when developing, I never saw a difference with docker-compose logs.

I will check tomorrow.

Reply:
Just to make sure we're not both sitting here waiting for the other to respond, you're still planning on looking into this right?

Reply:
Yes, I am planning to do this.

Reply:
So, I confirm that was is displayed by a `docker-compose up` in foreground is the same than a `docker logs -f boulder_boulder_1̀`. Which make sense, since both are supposed to display the stdout + stderr generated by the PID 1 of each Docker.

In fact Boulder is generated some logs after it started. Here is for instance, from `docker-compose up` through the `run_acme_server`, the output starting to be generated in the terminal while running a `certonly` certbot command against this local Boulder instance.

```
boulder_1    | All servers running. Hit ^C to kill.
boulder_1    | pebble-challtestsrv - 2019/06/30 10:55:22 Set default IPv4 address for DNS A queries to "10.77.77.1"
=> Finished boulder instance deployment.
--> Instance of http://localhost:4001/directory is running, directory URL is http://localhost:4001/directory
--> Press CTRL+C to stop the ACME server.
boulder_1    | I105947 boulder-va [AUDIT] Attempting to validate HTTP-01 for "test.example.com" with GET to "http://test.example.com/.well-known/acme-challenge/pqtx32L6Q2asy6EJmCuuXbpgOIRSsKRqK9y37jhaT2E"
boulder_1    | I105947 boulder-va [AUDIT] Checked CAA records for test.example.com, [Present: false, Account ID: 1, Challenge: http-01, Valid for issuance: true] Records=null
boulder_1    | I105947 boulder-va [AUDIT] Validation result JSON={"ID":"47jmyIhiPq74M6T6Ce5j1zHONEcKzCHUArr5SpSpaFQ","Requester":1,"Hostname":"test.example.com","Challenge":{"id":3,"type":"http-01","status":"valid","token":"pqtx32L6Q2asy6EJmCuuXbpgOIRSsKRqK9y37jhaT2E","keyAuthorization":"pqtx32L6Q2asy6EJmCuuXbpgOIRSsKRqK9y37jhaT2E.57dQskmRZbPyp-DKTD6t9pgj_Gx09oib1NlvwP2qY5Q","validationRecord":[{"url":"http://test.example.com/.well-known/acme-challenge/pqtx32L6Q2asy6EJmCuuXbpgOIRSsKRqK9y37jhaT2E","hostname":"test.example.com","port":"5002","addressesResolved":["10.77.77.1"],"addressUsed":"10.77.77.1"}]},"ValidationLatency":0.012}
boulder_1    | I105947 boulder-va Validations: {ID:47jmyIhiPq74M6T6Ce5j1zHONEcKzCHUArr5SpSpaFQ Identifier:{Type: Value:} RegistrationID:1 Status: Expires:<nil> Challenges:[] Combinations:[] Wildcard:false V2:false}
```

Reply:
Huh. That is indeed what [the docs](https://docs.docker.com/compose/reference/up/) say. When using docker-compose 1.17.1, this works for me, however, when I try using the latest version from PyPI I get no output from `docker-compose up`.

Do you see the same behavior? I don't see anything about this in [Docker Compose's changelog](https://github.com/docker/compose/blob/master/CHANGELOG.md), but is there a bug/undocumented change in behavior here?

Reply:
I did not have the time to check today, but maybe handling and solving this issue would be more suitable in #7120?

Reply:
I created https://github.com/certbot/certbot/issues/7216 to track this.